source/includes/api/_dataset.md

```diff
@@ -1053,20 +998,151 @@ curl -X POST https://api.resourcewatch.org/v1/dataset/:dataset_id/data-overwrite
+403            | Forbidden - User does not have access to this dataset's application | You specified an `application` in your request body that is not associated with your user account. 
+404            | Dataset with id <id> doesn't exist   | A dataset with the provided id does not exist.
+
+## Deleting a dataset
+
+Use this endpoint if you wish to delete a dataset. Deleting a dataset of type document (`connectorType` with value `document`) will cause the API's internal copy of said data to be deleted. Dataset types that proxy data from an external source (ie. carto, gee, etc) will be deleted without modifying said external source.
+
+When deleting a dataset that is associated with multiple `application` values, the user issuing the request must be associated with all of them in order for the dataset to be deleted. If that's not the case, the `application` values associated with the user will be removed from the dataset's `application` list, and no further action will be taken - the dataset itself and its associated resources will continue to exist. The dataset is only actually deleted if the user has access to all of the `application` to which the dataset belongs. 
+
+Besides deleting the dataset itself, this endpoint also deletes graph vocabularies, layers, widgets and metadata related to the dataset itself. These delete operations are issued in this order, and prior to deleting the dataset itself, but are not atomic - if one of them fails (for example, if attempting to delete a protected resource), the following ones are canceled, but the already deleted elements are not restored. 
+
+In order to delete a dataset, the following conditions must be met:
+
+- the dataset's `protected` property must be set to `false`.
+- the user must be logged in and belong to the same application as the dataset
+- the user must match one of the following:
+  - have role `ADMIN`
+  - have role `MANAGER` and be the dataset's owner (through the `userId` field of the dataset)
+
+
+> Example request for deleting a dataset
+
+```shell
+curl -X DELETE https://api.resourcewatch.org/v1/dataset/<dataset-id> \
+-H "Authorization: Bearer <your-token>"
+-H "Content-Type: application/json"
+```
+
+
+Error code     | Error message  | Description
+-------------- | -------------- | --------------
+400            | Dataset is protected | You are attempting to delete a dataset that has `protected` set to prevent deletion.
+401            | Unauthorized   | You need to be logged in to be able to delete a dataset.
+403            | Forbidden      | You need to either have the `ADMIN` role, or have role `MANAGER` and be the dataset's owner (through the `userId` field of the dataset)
+403            | Forbidden      | You are trying to delete a dataset with one or more `application` values that are not associated with your user account. 
+404            | Dataset with id <id> doesn't exist   | A dataset with the provided id does not exist.
+
+## Dataset automatic synchronization
```

Comment:
Is `sync` just an optional field in the dataset creation process? If so, I think this should be moved to the dataset creation section. Or it should at least be made clear that this is something you do at dataset creation time (or with a PATCH later).

Reply:
yeah, I thought about that too. My reasoning behind having this in a separate section is that it's an optional and complex feature. So, in the spirit of compartmentalizing this, as well as keeping the create section as short and simple as possible, I left this here.

Perhaps a good middle ground would be to have a "hey, we have this cool feature below, check it out if you are interested" link here from the create/update endpoints?

Reply:
Sure, that makes sense.

Reply:
added said notes to create, concat and overwrite docs.